File: src/exercicios/beecrowd/QuadroMedalhas/QuadroMedalhas.java
package exercicios.beecrowd.QuadroMedalhas;

import java.util.List;
import java.util.Scanner;
import java.util.ArrayList;

class Pais {
    public String nome;
    public int ouro;
    public int prata;
    public int bronze;

    public Pais(String nome, int ouro, int prata, int bronze) {
        this.nome = nome;
        this.ouro = ouro;
        this.prata = prata;
        this.bronze = bronze;
    }
}


public class QuadroMedalhas {

    public static void swap(List<Pais> paises, int i, int j) {
        Pais temp = paises.get(i);
        paises.set(i, paises.get(j));
        paises.set(j, temp);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int qtdPaises = sc.nextInt();

        List<Pais> paises = new ArrayList<>();

        for (int i = 0 ; i < qtdPaises; i++) {
            String nome = sc.next();
            int ouro = sc.nextInt();
            int prata = sc.nextInt();
            int bronze = sc.nextInt();
            paises.add(new Pais(nome, ouro, prata, bronze));
        }

        for (int i = 0; i < qtdPaises - 1; ++i) {
            for (int j = 0; j < qtdPaises - i - 1; j++) {
                if (paises.get(j).ouro < paises.get(j + 1).ouro) {
                    swap(paises, j, j + 1);
                } else if (paises.get(j).ouro == paises.get(j + 1).ouro) {
                    if (paises.get(j).prata < paises.get(j + 1).prata) {
                        swap(paises, j, j + 1);
                    } else if (paises.get(j).prata == paises.get(j + 1).prata) {
                        if (paises.get(j).bronze < paises.get(j + 1).bronze) {
                            swap(paises, j, j + 1);
                        } else if (paises.get(j).bronze == paises.get(j + 1).bronze) {
                            if (paises.get(j).nome.compareTo(paises.get(j + 1).nome) > 0) {
                                swap(paises, j, j + 1);
                            }
                        }
                    }
                }
            }
        }

        for (Pais pais : paises) {
            System.out.println(pais.nome + " " + pais.ouro + " " + pais.prata + " " + pais.bronze);
        }

        sc.close();
    }


}

File: src/exercicios/beecrowd/QuadroMedalhas/quadroMedalhas.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>


struct pais {
    char nome[100];
    int ouro;
    int prata;
    int bronze;
} typedef Pais;

struct quadro {
    Pais paises[100];
    int n;
} typedef Quadro;

void inicializaQuadro(Quadro *q) {
    q->n = 0;
}

void swap(Pais *a, Pais *b) {
    Pais aux = *a;
    *a = *b;
    *b = aux;
}

void adicionaPais(Quadro *q, char nome[], int ouro, int prata, int bronze) {
    strcpy(q->paises[q->n].nome, nome);
    q->paises[q->n].ouro = ouro;
    q->paises[q->n].prata = prata;
    q->paises[q->n].bronze = bronze;
    q->n++;
}

void ordenarQuadro(Quadro *q) {
    for (int i = 0; i < q->n; i++) {
        for ( int j = 0; j < q->n - i -1; j++){
            if (q->paises[j].ouro < q->paises[j+1].ouro) {
                swap(&q->paises[j], &q->paises[j+1]);
            } else if (q->paises[j].ouro == q->paises[j+1].ouro) {
                if (q->paises[j].prata < q->paises[j+1].prata) {
                    swap(&q->paises[j], &q->paises[j+1]);
                } else if (q->paises[j].prata == q->paises[j+1].prata) {
                    if (q->paises[j].bronze < q->paises[j+1].bronze) {
                        swap(&q->paises[j], &q->paises[j+1]);
                    } else if (q->paises[j].bronze == q->paises[j+1].bronze) {
                        if (strcmp(q->paises[j].nome, q->paises[j+1].nome) > 0) {
                            swap(&q->paises[j], &q->paises[j+1]);
                        }
                    }
                }
            }
        }
    }
}

int main(void) {
    Quadro q;
    inicializaQuadro(&q);

    while(!feof(stdin)) {
        char nome[100];
        int ouro, prata, bronze;
        scanf("%s %d %d %d", nome, &ouro, &prata, &bronze);
        adicionaPais(&q, nome, ouro, prata, bronze);
    }

    ordenarQuadro(&q);

    for (int i = 0; i < q.n; i++) {
        printf("%s %d %d %d\n", q.paises[i].nome, q.paises[i].ouro, q.paises[i].prata, q.paises[i].bronze);
    }

    return 0;
}